### ISSUE TITLE
'Is someone responding to bug reports' check failing with `AttributeError: 'str' object has no attribute 'get'`

### ISSUE THREAD

**Issue by matthewfeickert** [ryancheley/the-well-maintained-test#10]
**Describe the bug**
Hi. :wave: First, thanks for making this project &mdash; it seems like a cool idea in general and could be a useful tool for doing project/community health summaries (or first passes at least &mdash; software metrics are hard). :slightly_smiling_face: 

I'm seeing a small bug on check 4 (Is someone responding to bug reports?) when I apply it to some [Scikit-HEP](https://scikit-hep.org/) projects. As a particular failing example, I'll use [`pyhf`](https://github.com/scikit-hep/pyhf) as I maintain that.

**To Reproduce**

```console
$ docker run --rm -ti python:3.9 /bin/bash
root@16ee2e838cc6:/# python -m venv venv && . venv/bin/activate
(venv) root@16ee2e838cc6:/# python -m pip --quiet install --upgrade pip setuptools wheel
(venv) root@16ee2e838cc6:/# python -m pip --quiet install the-well-maintained-test
(venv) root@16ee2e838cc6:/# python -m pip list
Package                  Version
------------------------ ---------
certifi                  2021.10.8
charset-normalizer       2.0.8
click                    8.0.3
colorama                 0.4.4
commonmark               0.9.1
idna                     3.3
pip                      21.3.1
Pygments                 2.10.0
requests                 2.26.0
rich                     10.14.0
setuptools               59.2.0
the-well-maintained-test 0.3.1
urllib3                  1.26.7
wheel                    0.37.0
(venv) root@16ee2e838cc6:/# the-well-maintained-test 'https://github.com/scikit-hep/pyhf'
1. Is it described as 'production ready'?
        The project is set to Development Status Beta
2. Is there sufficient documentation?
        Documentation can be found at https://pyhf.readthedocs.io/
3. Is there a changelog?
        Yes
4. Is someone responding to bug reports?
Traceback (most recent call last):
  File "/venv/bin/the-well-maintained-test", line 8, in <module>
    sys.exit(cli())
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/the_well_maintained_test/cli.py", line 71, in cli
    print(bug_responding(bugs_url, headers))
  File "/venv/lib/python3.9/site-packages/the_well_maintained_test/utils.py", line 67, in bug_responding
    bug_create_date = datetime.strptime(i.get("created_at"), "%Y-%m-%dT%H:%M:%SZ")
AttributeError: 'str' object has no attribute 'get'
(venv) root@16ee2e838cc6:/# 
```

**Expected behavior**

For this to not fail and to give some output.

**Additional context**

I see that this is actually happening with the PyPI distributed version of `the-well-maintained-test` as well

```console
(venv) root@16ee2e838cc6:/# the-well-maintained-test 'https://github.com/ryancheley/the-well-maintained-test'
1. Is it described as 'production ready'?
        The project is set to Development Status Beta
2. Is there sufficient documentation?
        Documentation can be found at https://github.com/ryancheley/the-well-maintained-test/blob/main/README.md
3. Is there a changelog?
        Yes
4. Is someone responding to bug reports?
Traceback (most recent call last):
  File "/venv/bin/the-well-maintained-test", line 8, in <module>
    sys.exit(cli())
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/venv/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/venv/lib/python3.9/site-packages/the_well_maintained_test/cli.py", line 71, in cli
    print(bug_responding(bugs_url, headers))
  File "/venv/lib/python3.9/site-packages/the_well_maintained_test/utils.py", line 67, in bug_responding
    bug_create_date = datetime.strptime(i.get("created_at"), "%Y-%m-%dT%H:%M:%SZ")
AttributeError: 'str' object has no attribute 'get'
```

**Comment by ryancheley** [comment]
@matthewfeickert first, thanks for reporting the issue! I ran the code locally and you're correct, it's a rate limiting issue. 

If you add a file called `auth.json` that looks like this do you get the same error? 

```
{
    "github_personal_token": "github_personal_token goes here"
}
```

Thanks!

**Comment by ryancheley** [comment]
@matthewfeickert I've just released [version 0.4.0](https://pypi.org/project/the-well-maintained-test/0.4.0/) which does a couple of things to help resolve the issue. It's easier to generate the auth.json file now. Please let me know if the auth.json file fixes the issues for you. 

Thanks! 

Thanks!

**Comment by matthewfeickert** [comment]
Thanks @ryancheley, indeed, in `v0.4.0` things are resolved once I authenticate with my GitHub PAT. :+1: 

```console
$ docker run --rm -ti python:3.9 /bin/bash
root@6522daf89297:/# python -m venv venv && . venv/bin/activate
(venv) root@6522daf89297:/# python -m pip --quiet install --upgrade pip setuptools wheel
(venv) root@6522daf89297:/# python -m pip --quiet install 'the-well-maintained-test==0.4.0'
(venv) root@6522daf89297:/# python -m pip list
Package                  Version
------------------------ ---------
certifi                  2021.10.8
charset-normalizer       2.0.8
click                    8.0.3
colorama                 0.4.4
commonmark               0.9.1
idna                     3.3
pip                      21.3.1
Pygments                 2.10.0
requests                 2.26.0
rich                     10.15.0
setuptools               59.4.0
the-well-maintained-test 0.4.0
urllib3                  1.26.7
wheel                    0.37.0
(venv) root@6522daf89297:/# the-well-maintained-test --help
Usage: the-well-maintained-test [OPTIONS] COMMAND [ARGS]...

  Programatically tries to answer the 12 questions from         Adam Johnson's
  blog post https://adamj.eu/tech/2021/11/04/the-well-maintained-test/ URL is
  a url to a github repository you'd like to check, for example:     the-well-
  maintained-test 'https://github.com/ryancheley/the-well-maintained-test'

Options:
  --version  Show the version and exit.
  --help     Show this message and exit.

Commands:
  auth       Save authentication credentials to a JSON file
  questions  List of questions tested
  url        url to a github repository you'd like to check
(venv) root@6522daf89297:/# the-well-maintained-test auth
Create a GitHub personal user token and paste it here:

Personal token:
(venv) root@6522daf89297:/# file auth.json 
auth.json: JSON data
(venv) root@6522daf89297:/# the-well-maintained-test url 'https://github.com/scikit-hep/pyhf'
1. Is it described as 'production ready'?
        The project is set to Development Status Beta
2. Is there sufficient documentation?
        Documentation can be found at https://pyhf.readthedocs.io/
3. Is there a changelog?
        Yes
4. Is someone responding to bug reports?
        The maintainer took 0 days to respond to the bug report
        It has been 1322 days since a comment was made on the bug.
5. Are there sufficient tests?
        There are 43 tests:
                - src/pyhf/infer/test_statistics.py
                - tests/benchmarks/test_benchmark.py
                - tests/contrib/test_utils.py
                - tests/contrib/test_viz.py
                - tests/test_backend_consistency.py
                - tests/test_backends.py
                - tests/test_calculator.py
                - tests/test_cli.py
                - tests/test_combined_modifiers.py
                - tests/test_compat.py
                - tests/test_constraints.py
                - tests/test_custom_mods.py
                - tests/test_events.py
                - tests/test_examples.py
                - tests/test_export.py
                - tests/test_import.py
                - tests/test_infer.py
                - tests/test_init.py
                - tests/test_interpolate.py
                - tests/test_jit.py
                - tests/test_mixins.py
                - tests/test_modifiers.py
                - tests/test_notebooks.py
                - tests/test_optim.py
                - tests/test_paramsets.py
                - tests/test_paramviewer.py
                - tests/test_patchset.py
                - tests/test_pdf.py
                - tests/test_probability.py
                - tests/test_public_api.py
                - tests/test_public_api_repr.py
                - tests/test_regression.py
                - tests/test_schema.py
                - tests/test_scripts.py
                - tests/test_simplemodels.py
                - tests/test_tensor.py
                - tests/test_tensorviewer.py
                - tests/test_teststats.py
                - tests/test_toys.py
                - tests/test_utils.py
                - tests/test_validation.py
                - tests/test_workspace.py
                - validation/standard_hypo_test_demo.py

6. Are the tests running with the latest Language version?
        The project supports the following programming languages
                - Python 3
                - Python 3 :: Only
                - Python 3.7
                - Python 3.8
                - Python 3.9
                - Python Implementation :: CPython

7. Are the tests running with the latest Integration version?
        This project has no associated frameworks
8. Is there a Continuous Integration (CI) configuration?
        There are 18 workflows
         - CI/CD
         - Build Binder Image
         - Build Binder Image
         - CI/CD
         - CodeQL
         - HEAD of dependencies
         - dev-tools
         - Docker Images
         - Docs
         - Lint
         - Lint Code Base
         - .github/workflows/list_likelihoods.yml
         - Minimum supported dependencies
         - Merged PR
         - Notebooks
         - publish distributions
         - Current Release
         - Tag Creator

9. Is the CI passing?
        Yes
10. Does it seem relatively well used?
        The project has the following statistics:
        - Watchers: 196
        - Forks: 62
        - Open Issues: 340
        - Subscribers: 11
11. Has there been a commit in the last year?
        Yes. The last commit was on 11-24-2021 which was 4 days ago
12. Has there been a release in the last year?
        Yes. The last commit was on 09-03-2021 which was 86 days ago
(venv) root@6522daf89297:/# 
```
